### DISCUSSION TITLE
Soup error

### DISCUSSION THREAD

**Discussion by kvkamin** [Security-Onion-Solutions/securityonion#11091] [2.4]
errors from soup log. 

------------------------------------------
Looks like you have an updated OS
Let's see if we need to update Security Onion.
Checking to see if there are hotfixes needed
**We need to apply a hotfix**
You have enough space for upgrade. Proceeding with soup.
Checking for Salt Master and Minion updates.
You are already running the correct version of Salt for Security Onion.
 **hotfixg 
/sbin/soup: line 768: apply_hotfix: command not found**

Starting crond service at 21:37:27.307802
Successfully started crond.

Starting salt-master service at 21:37:27.318497
Successfully started salt-master.

Starting salt-minion service at 21:37:27.327732
Successfully started salt-minion.

Enabling highstate.
local:
    ----------
    msg:
        Info: highstate state already enabled.
    res:
        True

Soup failed with error 127: Unhandled error
Unhandled error occured, please check /root/soup.log for details.

**Comment by dougburks** [comment]
Please provide more information as shown in https://github.com/Security-Onion-Solutions/securityonion/discussions/1720:
> Please include the following details where you can:
Security Onion version as seen in the lower left corner of SOC and in /etc/soversion. For example: 2.4.10
Is this a cloud deployment or on-prem? If on-prem, do you have Internet access or this an airgap installation?
Did you install from our Security Onion ISO image or did you perform a network installation?
If network installation, what distro and version did you install on?

In addition, did you run `soup` by itself or did you provide any options?

**Comment by kvkamin** [comment]
sorry for lack of context.  
ran soup by hand, soup -y
version is 2.4.10
On prem deployment
Used Security Onion ISO image
Internet access system.

I originally tried installing Oracle 9.2 and then network setup because I wanted more control over disk slicing.  I have 8TB of raided nvme sticks that I wanted to mount as /nsm (super fast queries even with billions of records).  Then ran so-setup-network.  This setup had same soup error.  I knew there was probably no new version, I just wanted to see the output from soup.  I wondered if the soup error was from self install and network setup vs Security Onion ISO, so I reinstalled with ISO. Same error. To fix disk slicing requirements, after installing with ISO, at first reboot, I stopped install, created the nvme md raid, mounted it as /nsm2, moved all files that were installed in /nsm to /nsm2, remounted origninal /nsm slice as /data and /nsm2 as /nsm (of course in /etc/fstab) and rebooted.  

BTW, Security Onion is a very impressive system and it is important to me in my MSSP business.  Good work!

**Comment by dougburks** [comment]
What is the output of the following commands?
```
ls -alh /etc/sohotfix
```

```
cat /etc/sohotfix
```

**Reply by kvkamin** [reply]
[root@so-ka sbin]# ls -alh /etc/sohotfix
ls: cannot access '/etc/sohotfix': No such file or directory

Also, apply_hotfix function in /sbin/soup is commented out.

**Comment by dougburks** [comment]
> [root@so-ka sbin]# ls -alh /etc/sohotfix
> ls: cannot access '/etc/sohotfix': No such file or directory

That file should exist after installing our default 2.4.10 ISO image.

> Also, apply_hotfix function in /sbin/soup is commented out.

Yep, we've already fixed that in our dev branch. But the root issue here is that something has removed /etc/sohotfix on your box.

**Reply by kvkamin** [reply]
A little more info. I installed using Manager Search node on hardware, no virtualization.  I have done several installs, playing around with access mode options, ie ip, hostname, FQDN, and even moving to different circuit (we dropped a couple of Spectrum cable circuits that were under-performing).  Every single install had same soup issue.

**Reply by dougburks** [reply]
Really strange. All of our installations here have /etc/sohotfix. 

Are you installing any additional software or making any other modifications that could be removing /etc/sohotfix?

**Reply by kvkamin** [reply]
No additional software, after every install, ISO and so-setup-network (prolly  8 installs lolz) ran soup immediately after install. Always that error.

**Reply by dougburks** [reply]
If you are running soup IMMEDIATELY after install, then perhaps you're running before /etc/sohotfix has been created. 

Can you try a fresh install from the 2.4.10 ISO image and then wait 30 minutes after Setup completes to see if /etc/sohotfix appears?

**Reply by kvkamin** [reply]
Ah, ok, I will do a test on with a VM?  I'm pretty far along with my current install of 2.4, even deploying Elastic Agents with ManageEngine scripts and don't want to mess with current hardware install.

**Reply by CharlieMK** [reply]
I have the same problem, 
- ISO stand-alone install
- 2.4.10
-  full internet access 

exactly same error message. Sohotfix file does not exist. System was installed 2 days ago.

**Reply by kvkamin** [reply]
created vm .  First reboot, selected config network, /etc/sohotfix was not there. Did not run soup.  Rebooted, ran install routine, selected New, Distributed, Manager Search node.  Had another shell logged in running during setup:
 #watch ls /etc/so*
only /etc/soversion  appeared
After install finished, reboot, setup watch again. 
wait 30 minutes, still no /etc/sohotfix

**Reply by dougburks** [reply]
OK, thanks for the additional test. We have a hotfix coming out next week that should resolve all of this.
